src/LanguageServer/Impl/Implementation/Server.cs

```diff
@@ -624,22 +611,14 @@ private sealed class ReloadModulesQueueItem : IAnalyzable {
                 // since otherwise Complete() may come before the change is enqueued
                 // for processing and the completion list will be driven off the stale data.
                 var p = pending;
-                cookieTask.ContinueWith(t => {
-                    if (t.IsFaulted) {
-                        // Happens when file got deleted before processing
-                        p.Dispose();
-                        LogMessage(MessageType.Error, t.Exception.Message);
-                        return;
-                    }
-                    OnDocumentChangeProcessingComplete(doc, t.Result as VersionCookie, enqueueForAnalysis, priority, p);
-                }).DoNotWait();
+                await OnDocumentChangeProcessingCompleteAsync(doc, cookie as VersionCookie, enqueueForAnalysis, priority, p);
```

Comment:
Before the change, `EnqueueItem` has been completed before `ParseQueue.EnqueueAsync` was completed. Now it waits for the parse to finish. Also, before the change exceptions from `ParseQueue.Enqueue` were logged but didn't affect the caller. Right now, exception will be propagated up to the `DidChangeTextDocument`.

Reply:
Maybe it should be 

```cs
catch {
      pending?.Dispose();
      throw
  }
```
```